internal/states/encryption/example_test.go

```diff
@@ -0,0 +1,182 @@
+package encryption
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionconfig"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionflow"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+// Note: at the moment, we are still using MockUpLoggingFlow, which does not actually encrypt anything.
+
+func ExampleGetRemoteStateSingleton() {
+	// this is an example how the rest of tofu's codebase should interact with this package
+	// (case of our own remote state)
+
+	// before starting tofu, environment variables TF_STATE_ENCRYPTION and TF_STATE_DECRYPTION_FALLBACK
+	// are set by the user executing tofu. It is ok to not set either of them.
+
+	// Here is an example value that affects our own remote state (providing a passphrase):
+	TF_STATE_ENCRYPTION := `{
+      "backend": {
+		"key_provider": {
+			"config": {
+				"passphrase": "the current passphrase"
+			}
+		}
+	  }
+    }`
+	TF_STATE_DECRYPTION_FALLBACK := `{
+      "backend": {
+		"key_provider": {
+			"config": {
+				"passphrase": "the previous passphrase for key rotation"
+			}
+		}
+	  }
+    }`
```

Comment:
tiny nitpick: this is a bit hard to review in github due to the mixing of tabs and spaces.